packages/shacl-middleware/index.ts

```diff
@@ -74,6 +76,15 @@ export const shaclMiddleware = ({ getTargetNode, loadResource, loadResourcesType
     return next()
   }
 
+  if (resource.dataset.match(targetNode).size === 0 && shapes.match(null, sh.property).size > 0) {
+    const response = new ProblemDocument({
+      status: 400,
+      title: 'Request validation error',
+      detail: 'Missing target node in request body',
+    })
+    return res.status(400).send(response)
+  }
```

Comment:
Not too sure about this. I added this check to keep the same behavior.

Reply:
Mhm, so I added this check previously to prevent false positives when no validation actually occurred.

Why did it stop working after your changes?

Reply:
Because of this change in the lib https://github.com/zazuko/rdf-validate-shacl/pull/81

Reply:
Right. I wonder if SHACL should have some kind of "strict" mode so that both scenarios would natively supported

Reply:
🤷